### PR TITLE
feat(frontend): preview files instead of downloading it in comments

### PR DESCRIPTION
**Pull request by Millefeuille42**
#5997 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 1]
```diff
@@ -2,32 +2,118 @@ import React from 'react'
...
+import Lightbox from 'react-image-lightbox'
+import AttachedFile from '../AttachedFile/AttachedFile.jsx'
 import {
   buildFilePreviewUrl,
   removeExtensionOfFilename,
-  getFileDownloadUrl
+  getFileDownloadUrl, handleFetchResult
 } from '../../helper.js'
-import AttachedFile from '../AttachedFile/AttachedFile.jsx'
+import { getFileRevisionPreviewInfo } from '../../action.async'
```
Comment (grignards) [inline frontend_lib/src/component/Timeline/CommentFilePreview.jsx:12]: I think the usage is more to include the extension when importing a local file:
```suggestion
import { getFileRevisionPreviewInfo } from '../../action.async.js'
```

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 2]
```diff
@@ -49,22 +135,40 @@ export class CommentFilePreview extends React.Component {
...
+      <>
+        <a
+          className='CommentFilePreview'
+          title={title}
+          onClick={this.handleClickPreview}
+          href={fileDownloadUrl}
+          download
+        >
+          <AttachedFile fileName={filename} />
+          {(!this.state.fallbackPreview &&
```
Comment (grignards) [inline frontend_lib/src/component/Timeline/CommentFilePreview.jsx:147]:
```suggestion
          {(!state.fallbackPreview &&
```

**Review by grignards — APPROVED** [review]

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 3]
```diff
@@ -2,32 +2,118 @@ import React from 'react'
 import { translate } from 'react-i18next'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
+import Lightbox from 'react-image-lightbox'
+import AttachedFile from '../AttachedFile/AttachedFile.jsx'
 import {
   buildFilePreviewUrl,
   removeExtensionOfFilename,
-  getFileDownloadUrl
+  getFileDownloadUrl, handleFetchResult
```
Comment (gbonaspetti) [inline frontend_lib/src/component/Timeline/CommentFilePreview.jsx:10]:
```suggestion
  getFileDownloadUrl, 
  handleFetchResult
```

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 4]
```diff
@@ -2,32 +2,118 @@ import React from 'react'
...
 
     const workspaceId = props.apiContent.workspace_id || props.apiContent.workspaceId
     const filename = props.apiContent.filename || props.apiContent.fileName
     const id = props.apiContent.content_id || props.apiContent.id
     const currentRevisionId = props.apiContent.revision_id || props.apiContent.currentRevisionId
+    const filenameWithoutExtension = removeExtensionOfFilename(props.apiContent.filename)
+
+    const lightboxUrlList = (new Array(state.previewInfo.page_nb))
+      .fill(null)
+      .map((n, index) => // create an array [1..revision.page_nb]
```
Comment (gbonaspetti) [inline frontend_lib/src/component/Timeline/CommentFilePreview.jsx:105]:
```suggestion
      .map((n, index) => // INFO - ML - 2022-11-03 - create an array [1..revision.page_nb]
```

**Review by gbonaspetti — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch (the download button was missing, so Iadded on last commit)

**Review by Skylsmoi — COMMENTED** [review]
I'm a bit late sorry but I'd like this to be fixed please

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 5]
```diff
@@ -2,32 +2,119 @@ import React from 'react'
...
+
+  handleClickHideImageRaw = () => {
+    this.setState({ displayLightbox: false })
+  }
+
+  handleClickPreviousPage = () => {
+    const { state } = this
+    let page = state.fileCurrentPage
+
+    if (state.previewInfo.page_nb > 0) page--
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend_lib/src/component/Timeline/CommentFilePreview.jsx:75]: I'm going to fix during integration (#6023)

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx [hunk 6]
```diff
@@ -2,32 +2,119 @@ import React from 'react'
...
+
+    this.setState({
+      fileCurrentPage: page,
+      previewLoaded: false
+    })
+  }
+
+  handleClickNextPage = () => {
```
[the remark this answers is not in the document]
Comment (gbonaspetti) [reply frontend_lib/src/component/Timeline/CommentFilePreview.jsx:83]: I'm going to fix during integration (#6023)

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply frontend_lib/src/component/Timeline/CommentFilePreview.jsx:75]: Indeed this is really poorly done, looking back at it, I don't even know why I did it this way ^^'

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply frontend_lib/src/component/Timeline/CommentFilePreview.jsx:83]: Indeed this is really poorly done, looking back at it, I don't even know why I did it this way ^^'

File: frontend_lib/src/component/Timeline/CommentFilePreview.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Skylsmoi) [reply frontend_lib/src/component/Timeline/CommentFilePreview.jsx:75]: haha, maybe a bad day

**Pull request merged**